src/strategies/keystore.rs

```diff
@@ -0,0 +1,122 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+use tokio::sync::{Mutex, Notify};
```

Comment:
I think a plain `std::sync::Mutex` is more appropriate here?

from the [tokio doc](https://doc.servo.org/tokio/sync/struct.Mutex.html):
> The feature that the async mutex offers over the blocking mutex is the ability to keep it locked across an .await point. This makes the async mutex more expensive than the blocking mutex, so the blocking mutex should be preferred in the cases where it can be used. The primary use case for the async mutex is to provide shared mutable access to IO resources such as a database connection.

Given we are simply reading from a list in memory, blocking Mutex sounds fine? and should be cheaper